packages/medusa/src/api/admin/direct-upload/route.ts

```diff
@@ -0,0 +1,35 @@
+import { ulid } from "ulid"
+import { MIMEType } from "util"
+import type {
+  MedusaResponse,
+  AuthenticatedMedusaRequest,
+} from "@medusajs/framework/http"
+import type { HttpTypes } from "@medusajs/framework/types"
+import type { AdminDirectUploadRequestPayloadType } from "./validators"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminDirectUploadRequestPayloadType>,
+  res: MedusaResponse<HttpTypes.AdminDirectUploadSignatureResponse>
+) => {
+  const fileProvider = req.scope.resolve("file")
+  const { filename: originalname, mimeType, size } = req.body
+
+  const type = new MIMEType(mimeType)
+  const extension = type.subtype
+  const uniqueFilename = `${ulid()}.${extension}`
+
+  const response = await fileProvider.getUploadFileUrls({
+    filename: uniqueFilename,
+    mimeType: mimeType,
+    access: "private",
```

Comment:
Would this depend on what is being uploaded? Eg. for images you'd want them public

Reply:
Yeah, it depends how we want to categorize them. Should we treat all `images` as public and everything else as private?

Reply:
Maybe the caller should just pass if they want the upload to be private or public? But yes, all images are typically public

Reply:
+1 for a param, and default it to private if not present.